src/base/utils/enzymes.jl

```diff
@@ -8,7 +8,7 @@ function that returns a dictionary of solved fluxes.
 """
 gene_product_dict(model::GeckoModel, opt_model) =
     is_solved(opt_model) ?
-    Dict(genes(model) .=> value.(opt_model[:x])[(n_reactions(model)+1):end]) : nothing
+    Dict(genes(model) .=> value.(opt_model[:x])[(end-n_genes(model)+1):end]) : nothing
```

Comment:
avoid index subtraction at all cost.
```suggestion
    Dict(genes(model) .=> value.(opt_model[:x])[(length(model.columns)+1):end]) : nothing
```